src/Core/Checkout/Customer/SalesChannel/AbstractAccountService.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\SalesChannel;
+
+use Shopware\Core\Checkout\Customer\CustomerEntity;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+#[Package('checkout')]
+abstract class AbstractAccountService
```

Comment:
why do you need this abstract class to achieve your feature?

Reply:
Because there is currently no good way to override anything around the AccountService. This is not required to achieve the goal of this PR, but allows a developer to better override code or extend behavior after this would be merged.

Reply:
Removed the AbstractAccountService in AbstractAccountService